src/foam/u2/detail/SectionView.js

```diff
@@ -38,37 +55,74 @@ foam.CLASS({
       class: 'Boolean',
       name: 'showTitle',
       value: true
+    },
+    {
+      class: 'FObjectProperty',
+      of: 'foam.core.ArraySlot',
+      name: 'visibilitySlots_',
+      documentation: `
+        An array of slots used to control the visibility of the child
+        SectionedDetailPropertyViews of this view. Used internally to figure out
+        which children are the first and last ones visible so we can adjust the
+        padding on those elements to make sure nested
+        SectionedDetailPropertyViews don't cause unnecessary padding.
+      `,
+      factory: function() {
+        return this.ArraySlot.create();
+      }
+    },
+    {
+      class: 'Int',
+      name: 'firstVisibleIndex_',
+      expression: function(visibilitySlots_) {
+        return visibilitySlots_.get().indexOf(true);
+      }
+    },
+    {
+      class: 'Int',
+      name: 'lastVisibleIndex_',
+      expression: function(visibilitySlots_) {
+        return visibilitySlots_.get().lastIndexOf(true);
+      }
     }
   ],
   methods: [
     function initE() {
       var self = this;
-      this.addClass(this.myClass());
+      var slots = [];
       self.SUPER();
       
       self
+        .addClass(self.myClass())
         .add(self.slot(function(section, showTitle, section$title) {
           if ( ! section ) return;
           return self.Rows.create()
             .show(section.createIsAvailableFor(self.data$))
-            .callIf(showTitle && section$title, function () {
+            .callIf(showTitle && section$title, function() {
               this.start('h2').add(section$title).end();
             })
             .start(self.Grid)
-              .forEach(section.properties, function (p) {
-                this.start(self.GUnit, { 
-                  columns: p.gridColumns 
-                })
-                  .tag(self.SectionedDetailPropertyView, { prop: p, data$: self.data$ })
-                .end()
+              .forEach(section.properties, function(p, index) {
+                var slot = self.SimpleSlot.create({}, self);
```

Comment:
Maybe rather than doing this, can we put this inside the `start` (before your enableClass calls)
```
.call(function() { slots.push(this.visibilitySlot) })
```

Reply:
Resolved by 19661fbb0796bd0e1cc90ec1b74d1e9690800f03.